### PR TITLE
remove unused functions or mark as maybe_unused

### LINKED ISSUE #2414: Unused functions (dead code)
Running `cppcheck --enable=unusedFunction --force .` I discovered that we have quite a lot of unused functions (dead code) in our codebase. Should we remove (or comment out) all of them or are some of them still there for a purpose?

```
src/control/Control.cpp:2487:0: style: The function 'closeAndDestroy' is never used. [unusedFunction]

^
src/gui/PageView.cpp:808:0: style: The function 'containsY' is never used. [unusedFunction]

^
src/model/Stroke.cpp:389:0: style: The function 'debugPrint' is never used. [unusedFunction]

^
src/util/pixbuf-utils.cpp:89:0: style: The function 'f_image_surface_get_format' is never used. [unusedFunction]

^
src/util/pixbuf-utils.cpp:95:0: style: The function 'f_image_surface_get_height' is never used. [unusedFunction]

^
src/util/pixbuf-utils.cpp:93:0: style: The function 'f_image_surface_get_width' is never used. [unusedFunction]

^
src/control/xml/XmlAudioNode.cpp:9:0: style: The function 'getAudioFilepath' is never used. [unusedFunction]

^
src/pdf/base/XojPdfDocument.cpp:45:0: style: The function 'getDocumentInterface' is never used. [unusedFunction]

^
src/control/settings/Settings.cpp:1875:0: style: The function 'getDouble' is never used. [unusedFunction]

^
src/control/xojfile/LoadHandler.cpp:1036:0: style: The function 'getFileMinVersion' is never used. [unusedFunction]

^
src/util/audio/DeviceInfo.cpp:17:0: style: The function 'getInputChannels' is never used. [unusedFunction]

^
src/model/LinkDestination.cpp:68:0: style: The function 'getLeft' is never used. [unusedFunction]

^
src/control/RecentManager.cpp:89:0: style: The function 'getMaxRecent' is never used. [unusedFunction]

^
src/control/tools/EditSelectionContents.cpp:302:0: style: The function 'getOriginalHeight' is never used. [unusedFunction]

^
src/control/tools/EditSelectionContents.cpp:297:0: style: The function 'getOriginalWidth' is never used. [unusedFunction]

^
src/util/audio/DeviceInfo.cpp:19:0: style: The function 'getOutputChannels' is never used. [unusedFunction]

^
src/control/Control.cpp:2825:0: style: The function 'getRecentManager' is never used. [unusedFunction]

^
src/control/ToolHandler.cpp:313:0: style: The function 'getTools' is never used. [unusedFunction]

^
src/view/background/GraphBackgroundPainter.cpp:32:0: style: The function 'getUnitSize' is never used. [unusedFunction]

^
src/control/tools/EditSelection.cpp:254:0: style: The function 'getViewHeight' is never used. [unusedFunction]

^
src/control/tools/EditSelection.cpp:246:0: style: The function 'getViewWidth' is never used. [unusedFunction]

^
src/gui/widgets/gtkmenutooltogglebutton.cpp:153:0: style: The function 'gtk_menu_tool_toggle_button_class_init' is never used. [unusedFunction]

^
src/gui/widgets/gtkmenutooltogglebutton.cpp:467:0: style: The function 'gtk_menu_tool_toggle_button_get_menu' is never used. [unusedFunction]

^
src/gui/widgets/gtkmenutooltogglebutton.cpp:299:0: style: The function 'gtk_menu_tool_toggle_button_init' is never used. [unusedFunction]

^
src/gui/widgets/gtkmenutooltogglebutton.cpp:385:0: style: The function 'gtk_menu_tool_toggle_button_new_from_stock' is never used. [unusedFunction]

^
src/gui/widgets/gtkmenutooltogglebutton.cpp:501:0: style: The function 'gtk_menu_tool_toggle_button_set_arrow_tooltip_markup' is never used. [unusedFunction]

^
src/gui/widgets/gtkmenutooltogglebutton.cpp:484:0: style: The function 'gtk_menu_tool_toggle_button_set_arrow_tooltip_text' is never used. [unusedFunction]

^
src/model/Document.cpp:172:0: style: The function 'isPdfDocumentLoaded' is never used. [unusedFunction]

^
src/model/LinkDestination.cpp:26:0: style: The function 'link_dest_class_init' is never used. [unusedFunction]

^
src/model/LinkDestination.cpp:13:0: style: The function 'link_dest_init' is never used. [unusedFunction]

^
src/gui/MainWindow.cpp:568:0: style: The function 'reloadToolbars' is never used. [unusedFunction]

^
src/undo/EraseUndoAction.cpp:45:0: style: The function 'removeEdited' is never used. [unusedFunction]

^
src/control/RecentManager.cpp:78:0: style: The function 'removeRecentFileFilename' is never used. [unusedFunction]

^
src/gui/Redrawable.cpp:7:0: style: The function 'repaintElement' is never used. [unusedFunction]

^
src/gui/Redrawable.cpp:5:0: style: The function 'repaintRange' is never used. [unusedFunction]

^
src/gui/Redrawable.cpp:17:0: style: The function 'rerenderArea' is never used. [unusedFunction]

^
src/control/xml/XmlAudioNode.cpp:11:0: style: The function 'setAudioFilepath' is never used. [unusedFunction]

^
src/control/settings/Settings.cpp:1840:0: style: The function 'setComment' is never used. [unusedFunction]

^
src/control/RecentManager.cpp:91:0: style: The function 'setMaxRecent' is never used. [unusedFunction]

^
src/control/settings/Settings.cpp:1539:0: style: The function 'setPdfPageCacheSize' is never used. [unusedFunction]

^
src/control/xml/XmlStrokeNode.cpp:18:0: style: The function 'setPoints' is never used. [unusedFunction]

^
test/control/LoadHandlerTest.cpp:59:0: style: The function 'setUp' is never used. [unusedFunction]

^
src/model/LinkDestination.cpp:60:0: style: The function 'shouldChangeLeft' is never used. [unusedFunction]

^
src/model/LinkDestination.cpp:62:0: style: The function 'shouldChangeZoom' is never used. [unusedFunction]

^
src/model/Point.cpp:11:0: style: The function 'slopeTo' is never used. [unusedFunction]

^
src/gui/XournalView.cpp:84:0: style: The function 'staticLayoutPages' is never used. [unusedFunction]

^
test/control/LoadHandlerTest.cpp:61:0: style: The function 'tearDown' is never used. [unusedFunction]

^
test/util/XojPreviewExtractorTest.cpp:86:0: style: The function 'testLoad1Zipped' is never used. [unusedFunction]

^
test/util/XojPreviewExtractorTest.cpp:97:0: style: The function 'testLoad2Zipped' is never used. [unusedFunction]

^
test/control/LoadHandlerTest.cpp:74:0: style: The function 'testSpeedZipped' is never used. [unusedFunction]

^
src/util/logger/Logger.cpp:56:0: style: The function 'trace' is never used. [unusedFunction]

^
src/control/shaperecognizer/ShapeRecognizer.cpp:97:0: style: The function 'tryArrow' is never used. [unusedFunction]

^
src/gui/XournalView.cpp:480:0: style: The function 'zoomIn' is never used. [unusedFunction]

^
src/gui/XournalView.cpp:482:0: style: The function 'zoomOut' is never used. [unusedFunction]

^
src/gui/widgets/ZoomCallib.cpp:23:0: style: The function 'zoomcallib_class_init' is never used. [unusedFunction]

^
src/gui/widgets/ZoomCallib.cpp:36:0: style: The function 'zoomcallib_init' is never used. [unusedFunction]

^
```

### PR DESCRIPTION
**Pull request by rolandlo**
Closes #2414
I have also removed an unused variable in `Plugin.cpp`

### CODE REVIEW

**Review by Technius — APPROVED** [review]

**Comment by rolandlo** [comment]
One question though: According to various sources commenting out obsolete code is a bad thing. It should be rather removed and recovered from the version control system if it is ever needed again. So is it OK, if I remove all commented out code in the Shape recognizer?

**Comment by Febbe** [comment]
> One question though: According to various sources commenting out obsolete code is a bad thing. It should be rather removed and recovered from the version control system if it is ever needed again. So is it OK, if I remove all commented out code in the Shape recognizer?

Yes, you are right. But no-one indeed searches the version control system for previous written code. 
Maybe you should add a comment:
`// function Type xyz(Var var); removed after commit <current HEAD~1>`
@Technius what do you think?

File: src/control/settings/Settings.cpp [hunk 1]
```diff
@@ -1536,14 +1536,6 @@ void Settings::setPresentationHideElements(string elements) {
...
 auto Settings::getPdfPageCacheSize() const -> int { return this->pdfPageCacheSize; }
 
-void Settings::setPdfPageCacheSize(int size) {
-    if (this->pdfPageCacheSize == size) {
-        return;
-    }
-    this->pdfPageCacheSize = size;
-    save();
-}
-
```
Comment (Febbe) [inline src/control/settings/Settings.cpp:1546]: Is the pdfPageCacheSize obsolete? If yes, also remove the getter and the type. 
If not mark this function as [[maybe_unused]

Comment (rolandlo) [reply src/control/settings/Settings.cpp:1546]: No, it's not obsolete. The getter is still used in the `XournalView` class constructor and the `SidebarPreviewBase` constructor. So I will mark the getter as [[maybe_unused]].

**Comment by LittleHuba** [comment]
> Commenting them out will leave them in an unmaintained state and commenting them in could result in bugs.

That is exactly the reason why it should be removed. The reason you would want to keep it is to actually refer to the implementation later on when you reintroduce a feature. And at that point you either look it up in the VCS or just write it from scratch. But just having it commented out implies that it is usable as it is which might just not be the case.

@Febbe could you please also add this flag to the pipeline for PRs? I think we should mark this as failed test if somebody reintroduces unused methods. This keeps the code base cleaner.

**Comment by rolandlo** [comment]
This should be ready for being merged. If no objections are raised I will squash, rebase and merge tonight.

**Pull request merged**